examples/jetstream/api-migration/go/main.go

```diff
@@ -0,0 +1,255 @@
+	// a specific number of messages, without any optimization or prefetching.
+	// That makes using pull consumers in the legacy API inefficient in contrast
+	// to push consumers.
+	// In order to create a pull consumer and consume messages, the
+	// `PullSubscribe` method can be used.
+	fmt.Println("# Subscribe with pull consumer")
+	sub, _ = oldJS.PullSubscribe("events.>", "pull-cons", nats.AckExplicit())
+
+	// Messages can be retrieved using the `Fetch` or `FetchBatch` methods.
+	// `Fetch` will retrieve up to the provided number of messages and block
+	// until the either all messages are available or the timeout is reached.
+	fmt.Println("# Fetch")
+	msgs, _ := sub.Fetch(2, nats.MaxWait(100*time.Millisecond))
+	for _, msg := range msgs {
+		fmt.Printf("received %q\n", msg.Subject)
+		msg.Ack()
+	}
+
+	// `FetchBatch` is similar to `Fetch` except it will return a channel
+	// on which the messages will be delivered. This allows for processing
+	// messages as they become available.
+	fmt.Println("# FetchBatch")
+	msgs2, _ := sub.FetchBatch(2, nats.MaxWait(100*time.Millisecond))
+	for msg := range msgs2.Messages() {
+		fmt.Printf("received %q\n", msg.Subject)
+		msg.Ack()
+	}
+
+	// ## New JetStream API
+	// Now let's look at the new JetStream API for creating and managing
+	// streams and consumers.
+	fmt.Println("\n# New API")
+
+	// The new JetStream API is located in the `jetstream` package.
+	// In order to create an entry point to the JetStream API, use the
+	// `New` function.
+	newJS, _ := jetstream.New(nc)
+
+	// The new API uses `context.Context` for cancellation and timeouts when
+	// managing streams and consumers.
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// Creating a stream is done using the `CreateStream` method.
+	// It works similarly to the legacy `AddStream` method, except
+	// instead of returning `StreamInfo`, it returns a `Stream` handle,
+	// which can be used to manage the stream.
+	// Instead of creating a new stream, let's look up the existing `EVENTS` stream.
+	stream, _ := newJS.Stream(ctx, streamName)
+
+	// The new API differs from the legacy API in that it does not
+	// auto-create consumers. Instead, consumers must be created or retrieved
+	// explicitly. This allows for more control over the consumer lifecycle,
+	// while also getting rid of the hidden logic of the `Subscribe()` methods.
+	// In order to create a consumer, use the `AddConsumer` method.
+	// This method works similarly to the legacy `AddConsumer` method,
+	// except it returns a `Consumer` handle, which can be used to manage
+	// the consumer. Notice that since we are using pull consumers, we
+	// do not need to provide a `DeliverSubject`.
+	// In order to create a short-lived, ephemeral consumer, we will set the
+	// `InactivityThreshold` to a low value and not provide a consumer name.
+	cons, _ := stream.AddConsumer(ctx, jetstream.ConsumerConfig{
```

Comment:
So for streams, it's called `js.CreateStream` (per the above comment), but for consumers it is called `AddConsumer`? Either *add* or *create* should be standardized on across the board (including all clients).